core/src/main/kotlin/behavior/channel/StageChannelBehavior.kt

```diff
@@ -0,0 +1,76 @@
+package dev.kord.core.behavior.channel
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.cache.data.ChannelData
+import dev.kord.core.entity.channel.Channel
+import dev.kord.core.entity.channel.StageChannel
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+import dev.kord.rest.builder.channel.StageVoiceChannelModifyBuilder
+import dev.kord.rest.builder.guild.CurrentVoiceStateModifyBuilder
+import dev.kord.rest.builder.guild.VoiceStateModifyBuilder
+import dev.kord.rest.service.modifyCurrentVoiceState
+import dev.kord.rest.service.modifyVoiceState
+import dev.kord.rest.service.patchStageVoiceChannel
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+interface StageChannelBehavior : BaseVoiceChannelBehavior {
+
+    override fun withStrategy(
+        strategy: EntitySupplyStrategy<*>
+    ): StageChannelBehavior {
+        return StageChannelBehavior(id, guildId, kord, strategy.supply(kord))
+    }
+
+    fun StageChannelBehavior(
+        id: Snowflake,
+        guildId: Snowflake,
+        kord: Kord,
+        supplier: EntitySupplier = kord.defaultSupplier
+    ): StageChannelBehavior = object : StageChannelBehavior {
+        override val guildId: Snowflake get() = guildId
+        override val id: Snowflake get() = id
+        override val kord get() = kord
+        override val supplier get() = supplier
+    }
```

Comment:
This shouldn't be here

Reply:
Where should that be?

Reply:
outside of the interface ideally, but you already did so: https://github.com/kordlib/kord/pull/239/files/d346d5fd1c918c7d15f95e5d1f8b50df07cf85c7#diff-d6002ad60499d33441ee16a017fd1fcb2ce12050a4c65e56930839376f3c62d7R65-R76